### ISSUE TITLE
unable to detect channels outside of brain with `detect_bad_channels`

### ISSUE THREAD

**Issue by a-zmz** [SpikeInterface/spikeinterface#2929]
Hi, 

I tried to remove bad channels of my neuropixels 1.0 data, and the `detect_bad_channels` function could not detect any channels outside of brain no matter parameters...

key questions:

- since i shifted 48 channels to bank 1, channel 0-47 are actually at the top. in this case, what should i do with `outside_channels_location`?
- should i do this step on raw or highpass filtered data? with raw data, it can detect noise channels, but not on highpass filtered one. and yet by eye you can see those noisy channels that are outside of the brain:
![image](https://github.com/SpikeInterface/spikeinterface/assets/77285087/2bee0e3b-74a4-4144-acc2-b14fda63c7de)

ps. running spikeinterface v0.101.0 on linux fedora 39

Thanks!

**Comment by a-zmz** [comment]
hello @JoeZiminski, yea i realised the highpass filtering in the bad channel detection function. When passing raw to the function, i got those channels are labelled as `noisy`. And yea from now I would do the bad channel detection before bandpass filtering. Thanks very much!

re question about detecting `out_of_brain` channels if I have 48 channels shifted up:
- how should i set `outside_channels_location`?
- rather than `top` `bottom` or `both`, is it possible to use the actual depth position of channel for `outside_channels_location`?

**Comment by alejoe91** [comment]
The depths of the channels are used already, so top and bottom refer to the physical locations, not the channel indices ;)

**Comment by a-zmz** [comment]
oh right i see, then i should just use `top` if channels outside are further away from the shank tip?
But like i tried different options with the location, none of the channels were detected as `out_of_brain`, is there something else i could try?

**Comment by alejoe91** [comment]
yes `top` is correct. You can try to play around with the `outside_channel_threshold` parameter. @oliche any other suggestions?
